### PR TITLE
#164694229 Implement favoriting an article

### PR DESCRIPTION
**Pull request by Rhytah**
#### What does this PR do?
- Add favourite an article functionality

#### How this can this manually tests.
- Start up the server locally by running command `python manage.py runserver`
- Log in with verified credentials.
- to favourite an article use `PUT /api/articles/:slug/favorite/` endpoint
- to get favourited articles use `GET /api/articles/favorites/` endpoint
#### Related pivotal tracker stories.
[#164694229](https://www.pivotaltracker.com/story/show/164694229)

#### Relevant screenshots.
1. Fetch favorited articles

2. Favorite an article

3. Unfavorite an article

### CODE REVIEW

File: authors/apps/articles/migrations/0011_auto_20190425_2114.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Generated by Django 2.1.7 on 2019-04-25 18:14
+
```
Comment (kimbugp) [inline authors/apps/articles/migrations/0011_auto_20190425_2114.py]: @Rhytah , could you update your migrations files to only reflect your changes

File: authors/apps/articles/migrations/0010_auto_20190425_1925.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# Generated by Django 2.1.7 on 2019-04-25 16:25
+
+from django.conf import settings
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
```
Comment (kimbugp) [inline authors/apps/articles/migrations/0010_auto_20190425_1925.py]: @Rhytah , could you update your migrations files to only reflect your changes

File: authors/apps/articles/favourites_views.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
+import json
+
+from django.conf import settings
```
Comment (kimbugp) [inline authors/apps/articles/favourites_views.py]: @Rhytah, remove unused import

File: authors/apps/articles/favourites_views.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+from .models import Favourites, Article
+from .pagination import ArticleOffsetPagination
+from .renderers import FavortiesJsonRenderer
+from .serializers import FavouriteSerializer
+
+from authors.apps.profiles.models import Profile
+from authors.apps.authentication.models import User
+
+
+class FavouritesView(GenericAPIView):
```
Comment (kimbugp) [inline authors/apps/articles/favourites_views.py]: @Rhytah , add docstring

**Comment by kimbugp** [comment]
@Rhytah , rebase with develop

File: authors/apps/articles/urls.py [hunk 5]
```diff
@@ -56,4 +57,10 @@
         'bookmarks',
         BookMarkRetrieveAPI.as_view(),
         name='bookmarked'),
+    path(
```
Comment (kimbugp) [inline authors/apps/articles/urls.py:60]: @Rhytah, this url should only have method to create and update.
Don't allow list method

File: authors/apps/articles/favourites_views.py (hunk 4 above)
Comment (kimbugp) [inline authors/apps/articles/favourites_views.py]: @Rhytah, specify a more specific generic view

File: authors/apps/articles/favourites_views.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        return article_obj
+
+    def put(self, request, **kwargs):
+        article = self.get_object()
+        user = request.user
+        article_id = article.id
+        fav = Favourites.objects.filter(
+            article_id=article_id, user=user
+        ).first()
+        if fav:
```
Comment (kimbugp) [inline authors/apps/articles/favourites_views.py:48]: @Rhytah , could you refactor this logic a bit and also create a helper method which is documented

File: authors/apps/articles/favourites_views.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+    @classmethod
+    def get(self, request, *args, **kwargs):
+        user = request.user
+        queryset = Favourites.objects.filter(favourite=True, user=user)
+
+        serializer = FavouriteSerializer(
+            queryset, many=True, context={'request': request})
+
+        if not serializer.data:
```
Comment (kimbugp) [inline authors/apps/articles/favourites_views.py:89]: @Rhytah, could you add documentation of this implementations and why it's necessary

**Review by kimbugp — CHANGES_REQUESTED** [review]

File: authors/apps/articles/favourites_views.py (hunk 4 above)
Comment (Rhytah) [reply authors/apps/articles/favourites_views.py]: @kimbugp  I have specified a generic view to `Retrieve` and `Update` generic views

**Pull request closed without merging**